File: ledger-service/src/main/java/zachkingcade/dev/ledger/adapter/in/web/dto/accounttype/AccountTypeFilters.java
package zachkingcade.dev.ledger.adapter.in.web.dto.accounttype;

import java.util.List;
import java.util.Optional;

public record AccountTypeFilters(
        Optional<String> descriptionContains,
        Optional<String> notesContains,
        Optional<List<Long>> accountClass,
        Optional<Boolean> hideInactive,
        Optional<Boolean> hideActive
) {
}

File: user-interface-service/src/app/adapter/ledger-service/dto/account-types/AccountTypeFilters.ts
export interface IAccountTypeFilters {
  descriptionContains?: string;
  notesContains?: string;
  accountClass?: number[];
  hideInactive?: boolean;
  hideActive?: boolean;
  searchContains?: string;
  hideSystemAccounts?: boolean;
}


export type AccountTypeFilters = IAccountTypeFilters;
